sway-lib-std/src/vec.sw

```diff
@@ -142,3 +145,84 @@ impl<T> Vec<T> {
         self.len == 0
     }
 }
+
+/// TODO: Add proper errors
+
+pub struct StorageVec<V> {}
+
+impl<V> StorageVec<V> {
+    #[storage(read, write)]
+    fn push(self, value: V) {
+        let len = get::<u64>(__get_storage_key());
+        let key = sha256((len, __get_storage_key()));
+        store::<V>(key, value);
+        store(__get_storage_key(), len + 1);
+    }
+
+    #[storage(read, write)]
+    fn pop(self) -> Option<V> {
+        let len = get::<u64>(__get_storage_key());
+        if len == 0 {
+            return Option::None;
+        }
+    
+        let key = sha256((len, __get_storage_key()));
+        store(__get_storage_key(), len - 1);
+
+        Option::Some(get::<V>(key))
+    }
+
+    #[storage(read)]
+    fn get(self, index: u64) -> Option<V> {
+        let len = get::<u64>(__get_storage_key());
+        if len <= index {
+            return Option::None;
+        }
+
+        let key = sha256((index, __get_storage_key()));
+        Option::Some(get::<V>(key))
+    }
+
+    #[storage(read, write)]
+    fn remove_index(self, index: u64) -> Result<(), ()> {
```

Comment:
I'm unsure of the name `remove_index()`. Is that the same name that is used in Rust?

Reply:
Should this also return the item that has been removed?

Reply:
true, its actually just called remove, ill rename it as such (also it returns the removed element now)